cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -182,8 +186,8 @@ public boolean commitTx() throws Exception {
 
     // Publish a notification if there was new partition added.
     RuntimeProgramContext runtimeProgramContext = getRuntimeProgramContext();
-    if (runtimeProgramContext != null && !partitionsAddedInSameTx.isEmpty()) {
-      runtimeProgramContext.notifyNewPartitions(partitionsAddedInSameTx);
+    if (runtimeProgramContext != null && !operationsInThisTx.isEmpty()) {
+      runtimeProgramContext.notifyNewPartitions(Lists.transform(operationsInThisTx, OP_TO_KEY));
```

Comment:
not all of the `operationsInThisTx` are partition ADDs. Some of them are DELETEs.